In [120]:
import pandas as pd
import numpy as np

# Работа с Pandas

# Загрузите ваш DataFrame в переменную df методом pd.read_csv() и выведете на экран первые пять строк 

In [287]:
df = pd.read_csv('https://raw.githubusercontent.com/W1llAnn/1tdata-homework/main/1.3googleplaystore.csv', delimiter=',')
df.head()

App        Category  Rating   
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1  \
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type  Price Content Rating   
0     159   19M      10,000+  Free    0.0       Everyone  \
1     967   14M     500,000+  Free    0.0       Everyone   
2   87510  8.7M   5,000,000+  Free    0.0       Everyone   
3  215644   25M  50,000,000+  Free    0.0           Teen   
4     967  2.8M     100,000+  Free    0.0       Everyone   

                      Genres      Last Updated         Current Ver   
0               Art & Design   January 7, 2018               1.0.0  \
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

# Отобразите количество строк и столбцов в DataFrame

In [196]:
df.shape

(10841, 13)

**Видим, что в таблице 10841 строка и 13 столбцов**


 # Выведете основную информацию о DataFrame методом info(). Проанализируйте есть ли пропуски в столбцах. Напишите вывод о том, какие столбцы содержат пропуски

In [197]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  float64
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.1+ MB


**Пропуски есть в колонках: Rating, Type, Content Rating, Current Ver и Android Ver**

# Удалите или замените пропуски известными вам методами. (Подсказка: `dropna()` или `fillna()`, выбор того или иного метода должен быть аргументирован).

**Мы берем dropna, так как нам нужно удалить, а не заменить значения (как при fillna)**

In [172]:
# создадим копию и удалим все строки с NaN
df_drop = df.copy()
df_drop.dropna(inplace=True)

In [173]:
df_drop.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9360 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9360 non-null   object 
 1   Category        9360 non-null   object 
 2   Rating          9360 non-null   float64
 3   Reviews         9360 non-null   object 
 4   Size            9360 non-null   object 
 5   Installs        9360 non-null   object 
 6   Type            9360 non-null   object 
 7   Price           9360 non-null   float64
 8   Content Rating  9360 non-null   object 
 9   Genres          9360 non-null   object 
 10  Last Updated    9360 non-null   object 
 11  Current Ver     9360 non-null   object 
 12  Android Ver     9360 non-null   object 
dtypes: float64(2), object(11)
memory usage: 1023.8+ KB


In [132]:
print(f'Мы убрали примерно {round(100 - (9360 / 10841 * 100) , 3)} % из выборки, что можно считать большим количеством')

Мы убрали примерно 13.661 % из выборки, что можно считать большим количеством


**Или можем попробовать заменить значения, повторим пропущенные значения: Rating, Type, Content Rating, Current Ver и Android Ver**
**И посмотрим на уникальные значения в каждом столбце:**

**Rating**

In [161]:
df['Rating'].unique()

array([ 4.1,  3.9,  4.7,  4.5,  4.3,  4.4,  3.8,  4.2,  4.6,  3.2,  4. ,
        nan,  4.8,  4.9,  3.6,  3.7,  3.3,  3.4,  3.5,  3.1,  5. ,  2.6,
        3. ,  1.9,  2.5,  2.8,  2.7,  1. ,  2.9,  2.3,  2.2,  1.7,  2. ,
        1.8,  2.4,  1.6,  2.1,  1.4,  1.5,  1.2, 19. ])

**В столбце Rating больше всего пропущенных значений: 9367 (тип float64)**

***Type***

In [134]:
df['Type'].unique()

array(['Free', 'Paid', nan, '0'], dtype=object)

**В столбце Type есть 1 пропущенное и 1 "0"**

In [135]:
df.replace(['0'], [None], inplace=True)

In [136]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10245 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10840 non-null  object 
 6   Type            10839 non-null  object 
 7   Price           10841 non-null  float64
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.1+ MB


In [137]:
df['Type'].unique()

array(['Free', 'Paid', nan, None], dtype=object)

***Content Rating***

In [138]:
df['Content Rating'].unique()

array(['Everyone', 'Teen', 'Everyone 10+', 'Mature 17+',
       'Adults only 18+', 'Unrated', nan], dtype=object)

**Content Rating так же имеет только 1 пропущенное значение**

***Current Ver***

In [139]:
df['Current Ver'].unique()

array(['1.0.0', '2.0.0', '1.2.4', ..., '1.0.612928', '0.3.4', '2.0.148.0'],
      dtype=object)

**Current Ver - 8 пропущенных значений**

***Android Ver***

In [140]:
df['Android Ver'].unique()

array(['4.0.3 and up', '4.2 and up', '4.4 and up', '2.3 and up',
       '3.0 and up', '4.1 and up', '4.0 and up', '2.3.3 and up',
       'Varies with device', '2.2 and up', '5.0 and up', '6.0 and up',
       '1.6 and up', '1.5 and up', '2.1 and up', '7.0 and up',
       '5.1 and up', '4.3 and up', '4.0.3 - 7.1.1', '2.0 and up',
       '3.2 and up', '4.4W and up', '7.1 and up', '7.0 - 7.1.1',
       '8.0 and up', '5.0 - 8.0', '3.1 and up', '2.0.1 and up',
       '4.1 - 7.1.1', nan, '5.0 - 6.0', '1.0 and up', '2.2 - 7.1.1',
       '5.0 - 7.1.1'], dtype=object)

**Android Ver - 3 пропущенных значения**

**Сначала попробуем заменить только Rating, например на средний рейтинг по всем значениям. Используем fillna()**

In [174]:
# создадим копию, заполним Rating и удалим все строки с NaN
df_fil = df.copy()
df_fil['Rating'] = df_fil['Rating'].fillna(df['Rating'].mean())

**И удалим пропущенные значения**

In [175]:
df_fil.dropna(inplace=True)

In [176]:
df_fil.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10829 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10829 non-null  object 
 1   Category        10829 non-null  object 
 2   Rating          10829 non-null  float64
 3   Reviews         10829 non-null  object 
 4   Size            10829 non-null  object 
 5   Installs        10829 non-null  object 
 6   Type            10829 non-null  object 
 7   Price           10829 non-null  float64
 8   Content Rating  10829 non-null  object 
 9   Genres          10829 non-null  object 
 10  Last Updated    10829 non-null  object 
 11  Current Ver     10829 non-null  object 
 12  Android Ver     10829 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.2+ MB


In [156]:
print(f'Мы убрали примерно {round(100 - (10829 / 10841 * 100) , 3)} % из выборки, по сравнению с dropna значение сильно уменьшилось')

Мы убрали примерно 0.111 % из выборки, по сравнению с dropna значение сильно уменьшилось


# Проверьте нет ли пропусков после произведенной обработки

In [157]:
df_drop.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9360 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9360 non-null   object 
 1   Category        9360 non-null   object 
 2   Rating          9360 non-null   float64
 3   Reviews         9360 non-null   object 
 4   Size            9360 non-null   object 
 5   Installs        9360 non-null   object 
 6   Type            9360 non-null   object 
 7   Price           9360 non-null   float64
 8   Content Rating  9360 non-null   object 
 9   Genres          9360 non-null   object 
 10  Last Updated    9360 non-null   object 
 11  Current Ver     9360 non-null   object 
 12  Android Ver     9360 non-null   object 
dtypes: float64(2), object(11)
memory usage: 1023.8+ KB


In [158]:
df_fil.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10829 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10829 non-null  object 
 1   Category        10829 non-null  object 
 2   Rating          10829 non-null  float64
 3   Reviews         10829 non-null  object 
 4   Size            10829 non-null  object 
 5   Installs        10829 non-null  object 
 6   Type            10829 non-null  object 
 7   Price           10829 non-null  float64
 8   Content Rating  10829 non-null  object 
 9   Genres          10829 non-null  object 
 10  Last Updated    10829 non-null  object 
 11  Current Ver     10829 non-null  object 
 12  Android Ver     10829 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.2+ MB


**В обоих случаях все пропуски были убраны из DataFrame**

# 6. Выведите на экран минимум и максимум из столбца `Price`:

In [292]:
max_price = df['Price'].max()
print(f'максимум из столбца {max_price}')

min_price = df['Price'].min()
print(f'минимум из столбца {min_price}')

максимум из столбца 400.0
минимум из столбца 0.0


# Измените тип данных столбца Price на числовой с помощью метода `.astype()` :

In [159]:
df['Price'].unique()

array([  0.  ,   4.99,   3.99,   6.99,   1.49,   2.99,   7.99,   5.99,
         3.49,   1.99,   9.99,   7.49,   0.99,   9.  ,   5.49,  10.  ,
        24.99,  11.99,  79.99,  16.99,  14.99,   1.  ,  29.99,  12.99,
         2.49,  10.99,   1.5 ,  19.99,  15.99,  33.99,  74.99,  39.99,
         3.95,   4.49,   1.7 ,   8.99,   2.  ,   3.88,  25.99, 399.99,
        17.99, 400.  ,   3.02,   1.76,   4.84,   4.77,   1.61,   2.5 ,
         1.59,   6.49,   1.29,   5.  ,  13.99, 299.99, 379.99,  37.99,
        18.99, 389.99,  19.9 ,   8.49,   1.75,  14.  ,   4.85,  46.99,
       109.99, 154.99,   3.08,   2.59,   4.8 ,   1.96,  19.4 ,   3.9 ,
         4.59,  15.46,   3.04,   4.29,   2.6 ,   3.28,   4.6 ,  28.99,
         2.95,   2.9 ,   1.97, 200.  ,  89.99,   2.56,  30.99,   3.61,
       394.99,   1.26,   1.2 ,   1.04])

**Столбец Price имеет тип float64, но если бы нам нужно было менять его тип: df["Price"].astype('float')**

**А вот для следующего пункта нужно будет преобразовать Reviews, который имеет тип object**

In [12]:
df['Reviews'].unique()

array(['159', '967', '87510', ..., '603', '1195', '398307'], dtype=object)

In [178]:
df_fil = df_fil.astype({'Reviews': np.int8})
df_drop = df_drop.astype({'Reviews': np.int8})

In [164]:
df_fil.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10829 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10829 non-null  object 
 1   Category        10829 non-null  object 
 2   Rating          10829 non-null  float64
 3   Reviews         10829 non-null  int8   
 4   Size            10829 non-null  object 
 5   Installs        10829 non-null  object 
 6   Type            10829 non-null  object 
 7   Price           10829 non-null  float64
 8   Content Rating  10829 non-null  object 
 9   Genres          10829 non-null  object 
 10  Last Updated    10829 non-null  object 
 11  Current Ver     10829 non-null  object 
 12  Android Ver     10829 non-null  object 
dtypes: float64(2), int8(1), object(10)
memory usage: 1.1+ MB


In [165]:
df_drop.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9360 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9360 non-null   object 
 1   Category        9360 non-null   object 
 2   Rating          9360 non-null   float64
 3   Reviews         9360 non-null   int8   
 4   Size            9360 non-null   object 
 5   Installs        9360 non-null   object 
 6   Type            9360 non-null   object 
 7   Price           9360 non-null   float64
 8   Content Rating  9360 non-null   object 
 9   Genres          9360 non-null   object 
 10  Last Updated    9360 non-null   object 
 11  Current Ver     9360 non-null   object 
 12  Android Ver     9360 non-null   object 
dtypes: float64(2), int8(1), object(10)
memory usage: 959.8+ KB


# Выведите на экран медиану и среднее арифметическое столбцов:
# - Rating
# - Reviews

**Сначала посмотрим df_drop, где мы просто удалили все пропуски**

In [166]:
print(f" Среднее арифметическое в df_drop Rating: {df_drop['Rating'].mean()}")
print(f" Медиана в df_drop Rating {df_drop['Rating'].median()}")

print(f" Среднее в df_drop арифметическое Reviews: {df_drop['Reviews'].mean()}")
print(f" Медиана в df_drop Reviews {df_drop['Reviews'].median()}")

 Среднее арифметическое в df_drop Rating: 4.191837606837606
 Медиана в df_drop Rating 4.3
 Среднее в df_drop арифметическое Reviews: 4.5479700854700855
 Медиана в df_drop Reviews 10.0


**Ну и для себя посчитаем тоже самое и для Price**

In [167]:
print(f" Среднее в df_drop арифметическое Price: {df_drop['Price'].mean()}")
print(f" Медиана в df_drop Price {df_drop['Price'].median()}")

 Среднее в df_drop арифметическое Price: 0.9612788461538462
 Медиана в df_drop Price 0.0


**Теперь посмотрим df_fil, где мы просто удалили все пропуски**

In [169]:
print(f" Среднее арифметическое в df_fil Rating: {df_fil['Rating'].mean()}")
print(f" Медиана в df_fil Rating {df_fil['Rating'].median()}")

print(f" Среднее в df_fil арифметическое Reviews: {df_fil['Reviews'].mean()}")
print(f" Медиана в df_fil Reviews {df_fil['Reviews'].median()}")

 Среднее арифметическое в df_fil Rating: 4.1920411843445775
 Медиана в df_fil Rating 4.2
 Среднее в df_fil арифметическое Reviews: 4.611690830178225
 Медиана в df_fil Reviews 5.0


In [170]:
print(f" Среднее в df_fil арифметическое Price: {df_fil['Price'].mean()}")
print(f" Медиана в df_fil Price {df_fil['Price'].median()}")

 Среднее в df_fil арифметическое Price: 1.028091236494598
 Медиана в df_fil Price 0.0


**Ради интереса сравним с общей выборкой, например по Rating и Price**

In [179]:
print(f" Среднее арифметическое в df Rating: {df['Rating'].mean()}")
print(f" Медиана в df Rating {df['Rating'].median()}")


print(f" Среднее в df арифметическое Price: {df['Price'].mean()}")
print(f" Медиана в df Price {df['Price'].median()}")

 Среднее арифметическое в df Rating: 4.193338315362443
 Медиана в df Rating 4.3
 Среднее в df арифметическое Price: 1.0272733142699015
 Медиана в df Price 0.0


**Среднее арифметическое Rating и медиана Rating в общей выборке, df_drop и df_fil отличаются незначительно**
**А вот по Price есть отличия: к показателю общей выборки ближе показатель df_fil, т.е. выборки где мы сохранили практически все строки**

# Выведите на экран все уникальные значения категориального столбца Genres

In [198]:
df['Genres'].unique()

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

In [199]:
df['Genres'].nunique()

120

**120 уникальных значений**

# Сгруппируйте данные по столбцу `Genres` и посчитайте для каждого жанра средний и медианный рейтинг. Представьте результат в виде нового DataFrame,который будет сохранен в переменную `grouped_df`, где одна колонка будет содержать названия жарнов, а две другие буду содержать в себе средний и медианный рейтинги. Напишите краткий вывод.

In [248]:
grouped_df = df.groupby(['Genres']).agg({'Rating': ['mean', 'median']}).reset_index()
grouped_df.columns = ['Genres', 'mean', 'median']


grouped_df


Genres      mean  median
0                                   Action  4.285475     4.3
1                Action;Action & Adventure  4.311765     4.3
2                                Adventure  4.180822     4.3
3             Adventure;Action & Adventure  4.423077     4.5
4                    Adventure;Brain Games  4.600000     4.6
..                                     ...       ...     ...
115                Video Players & Editors  4.063924     4.2
116     Video Players & Editors;Creativity  4.100000     4.1
117  Video Players & Editors;Music & Video  4.000000     4.0
118                                Weather  4.244000     4.3
119                                   Word  4.410714     4.4

[120 rows x 3 columns]

In [251]:
grouped_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Genres  120 non-null    object 
 1   mean    116 non-null    float64
 2   median  116 non-null    float64
dtypes: float64(2), object(1)
memory usage: 2.9+ KB


# Из получившегося датафрейма выведите на экран только те жанры, чей медианный рейтинг больше 4.5. Напишите краткий вывод.

In [253]:
df_filtered = grouped_df[grouped_df['median'] > 4.5]
df_filtered

Genres       mean  median
4          Adventure;Brain Games   4.600000     4.6
11       Art & Design;Creativity   4.400000     4.7
18            Board;Pretend Play   4.800000     4.8
35             Comics;Creativity   4.800000     4.8
55      Entertainment;Creativity   4.533333     4.6
60             February 11, 2018  19.000000    19.0
65    Health & Fitness;Education   4.700000     4.7
75           Music;Music & Video   4.533333     4.6
88              Puzzle;Education   4.600000     4.6
106  Strategy;Action & Adventure   4.600000     4.6

In [262]:
df_filtered.shape

(10, 3)

**Из 120 Genres после фильтрации по медиане > 4.5 осталость всего 10 Genres**

**Выведете кол-во дубликатов в основном DataFrame (df) и удалите их**

In [257]:
len(df)- len(df.drop_duplicates ())

483

**483 дубликата по строкам, удалим их:**

In [258]:
df = df.drop_duplicates() 

In [259]:
len(df)- len(df.drop_duplicates ())

0

**Теперь дубликатов 0**

In [260]:
df.shape

(10358, 13)

**Из общей выборки осталось 10358 строк**

# Посчитатайте какое количество игр содержится в каждом жанре (value_counts). Выведите на экран топ-10 игр по количеству в жанре. Напишите краткий вывод.

In [275]:
df['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

In [273]:
game_df = df[df['Category'] == 'GAME']

game_df

App Category  Rating   Reviews Size   
1653                         ROBLOX     GAME     4.5   4447388  67M  \
1654                 Subway Surfers     GAME     4.5  27722264  76M   
1655               Candy Crush Saga     GAME     4.4  22426677  74M   
1656                      Solitaire     GAME     4.7    254258  23M   
1657                 Bubble Shooter     GAME     4.5    148897  46M   
...                             ...      ...     ...       ...  ...   
10791             Winter Wonderland     GAME     4.0      1287  38M   
10792        Soccer Clubs Logo Quiz     GAME     4.2     21661  16M   
10793                     Sid Story     GAME     4.4     28510  78M   
10803  Fatal Raid - No.1 Mobile FPS     GAME     4.3     56496  81M   
10804                  Poker Pro.Fr     GAME     4.2      5442  17M   

             Installs  Type  Price Content Rating   
1653     100,000,000+  Free    0.0   Everyone 10+  \
1654   1,000,000,000+  Free    0.0   Everyone 10+   
1655     500,000,000+  Free    0.0       Everyone   
1656      10,000,000+  Free    0.0       Everyone   
1657      10,000,000+  Free    0.0       Everyone   
...               ...   ...    ...            ...   
10791         50,000+  Free    0.0       Everyone   
10792      1,000,000+  Free    0.0       Everyone   
10793        500,000+  Free    0.0           Teen   
10803      1,000,000+  Free    0.0           Teen   
10804        100,000+  Free    0.0           Teen   

                             Genres       Last Updated   Current Ver   
1653   Adventure;Action & Adventure      July 31, 2018  2.347.225742  \
1654                         Arcade      July 12, 2018        1.90.0   
1655                         Casual       July 5, 2018     1.129.0.2   
1656                           Card     August 1, 2018       2.137.0   
1657                         Casual      July 17, 2018        1.20.1   
...                             ...                ...           ...   
10791                          Word  December 18, 2013           1.0   
10792                        Trivia       May 24, 2018        1.3.81   
10793                          Card     August 1, 2018         2.6.6   
10803                        Action     August 7, 2018       1.5.447   
10804                          Card       May 22, 2018         4.1.3   

        Android Ver  
1653     4.1 and up  
1654     4.1 and up  
1655     4.1 and up  
1656     4.1 and up  
1657   4.0.3 and up  
...             ...  
10791    2.2 and up  
10792    4.0 and up  
10793  4.0.3 and up  
10803    4.0 and up  
10804    2.3 and up  

[1144 rows x 13 columns]

In [274]:
game_df.shape

(1144, 13)

**Всего 1144 строки, теперь перейдем к подсчету, какое количество игр содержится в каждом жанре**

In [282]:
game5_df = game_df['Genres'].value_counts()
game5_df.head(5)

Genres
Action       365
Arcade       220
Racing        98
Adventure     75
Card          48
Name: count, dtype: int64

In [285]:
print(f'Больше всего игр в жанре Action 365 или {round((365 / 1144 * 100) , 3)} % от общего количества')
print(f'Второе место у жанра Arcade 220 или {round((220 / 1144 * 100) , 3)} % от общего количества')
print(f'В жанре Racing 98 или {round((98 / 1144 * 100) , 3)} % от общего количества')
print(f'В жанре Adventure 75 или {round((75 / 1144 * 100) , 3)} % от общего количества')
print(f'В жанре Card 48 или {round((48 / 1144 * 100) , 3)} % от общего количества')

Больше всего игр в жанре Action 365 или 31.906 % от общего количества
Второе место у жанра Arcade 220 или 19.231 % от общего количества
В жанре Racing 98 или 8.566 % от общего количества
В жанре Adventure 75 или 6.556 % от общего количества
В жанре Card 48 или 4.196 % от общего количества
